### PR TITLE
fix(eslint-plugin): [no-array-constructor] remove optional chaining exemption

### LINKED ISSUE #10933: Bug: [no-array-constructor] Does not detect `Array` call with optional chaining
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=IIJxEME8AoEoG4BQjQUgfgHR3kA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAQ2mmIE9CBjfXZAFzmof2nSnwDMvIAacNkgABBhQAOKatACW4hoRTwZuBgHoCJMpRp1GzVuwyREZNpEEBfEJaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Array();

Array?.();
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "no-array-constructor": "off",
    "@typescript-eslint/no-array-constructor": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`Array?.()` should be reported the same way `Array()` is.

### Actual Result

No report.

### Additional Info

`Array?.()` is reported in the core rule: https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5KCk7XG5cbkFycmF5Py4oKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

💖

### PR DESCRIPTION
**Pull request by developer-bandi**
## PR Checklist

- [x] Addresses an existing open issue: fixes #10933
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

remove optional chaining logic and change test case

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-array-constructor.ts [hunk 1]
```diff
@@ -43,7 +42,10 @@ export default createRule({
               return fixer.replaceText(node, '[]');
             }
             const fullText = context.sourceCode.getText(node);
-            const preambleLength = node.callee.range[1] - node.range[0];
+            const preambleLength =
+              node.parent.type === AST_NODE_TYPES.ChainExpression
+                ? node.callee.range[1] + 2 - node.range[0]
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-array-constructor.ts:47]: > ` + 2`

[Testing] 😬 Hardcoded lengths like `'?'.length` & equivalent are a bit dangerous to rely on. They tend to be broken over time by AST oddities like comments and whitespace. Since this rule logic is newly dependent on character lengths, could you please add some tests to make sure this is working as intended in those edge cases?

Here's a "fun" one: 

```ts
/* a */ /* b */ Array /* c */ /* d */ ?. /* e */ /* f */ () /* g */ /* h */
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Progress 💪

File: packages/eslint-plugin/src/rules/no-array-constructor.ts (hunk 1 above)
Comment (developer-bandi) [reply packages/eslint-plugin/src/rules/no-array-constructor.ts]: Added test code by referencing upstream and improved it to avoid hardcoding.

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks! 🚀 

I'll note that the fixers do drop some comments, but that's an existing issue. Filed: #11013
